### PR TITLE
SEP-31 support selecting customer types

### PR DESCRIPTION
**Pull request by quietbits**
- Updated `sep31Send` flow to add an extra step for selecting customer type (if there are multiple options available).
- Added `CAN_PROCEED` status to `ActionStatus`, which is similar to `SUCCESS` but happens during the process (`SUCCESS` should happen only once at the end of the process).
- Added `RadioButton` component.
- Added a modal to select sender/receiver types.

image

### CODE REVIEW

File: src/components/RadioButton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import "./styles.scss";
+
+interface RadioButtonProps extends React.InputHTMLAttributes<HTMLInputElement> {
+  id: string;
+  label: string | React.ReactNode;
+}
+
+export const RadioButton = ({ id, label, ...props }: RadioButtonProps) => (
+  <div className="RadioButton">
+    <input type="radio" id={id} {...props} />
```
Comment (JakeUrban) [inline src/components/RadioButton/index.tsx:11]: ❓ Will `{...props}` render with the correct HTML tag syntax?

File: src/components/Sep31Send.tsx [hunk 2]
```diff
@@ -49,76 +84,185 @@ export const Sep31Send = () => {
...
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    setErrorMessage("");
+    const { sender, receiver } = customerTypes;
+    event.preventDefault();
+
+    if (!sender || !receiver) {
+      setErrorMessage("Sender and receiver types are required");
+      return;
+    }
```
Comment (JakeUrban) [inline src/components/Sep31Send.tsx:113]: A receiving anchor could only require KYC for one of the sending & receiving customers, or neither.

**Review by JakeUrban — CHANGES_REQUESTED** [review]
I only found one issue, which is that we need to adjust the UI depending on whether or not the receiving anchor requires KYC for the sender, receiver, or both.

File: src/components/RadioButton/index.tsx (hunk 1 above)
Comment (quietbits) [reply src/components/RadioButton/index.tsx:11]: Yes, all the native input props will render correctly (name, value, etc).

File: src/components/Sep31Send.tsx (hunk 2 above)
Comment (quietbits) [reply src/components/Sep31Send.tsx:113]: I'll add better validation 👍

**Review by JakeUrban — APPROVED** [review]
LGTM

**Pull request merged**